src/test/java/io/pravega/connectors/hadoop/PravegaConnectorLocalJobITCase.java

```diff
@@ -65,29 +78,99 @@ public void setUp() throws Exception {
+
+        fs = FileSystem.getLocal(conf);
         if (fs.exists(outputPath)) {
             fs.delete(outputPath, true);
         }
-    }
 
-    @Test
-    public void testPravegaConnector() throws Exception {
-        boolean status = job.waitForCompletion(true);
+        String endPos1 = PravegaInputFormat.fetchLatestPositionsJson(
+                SETUP_UTILS.getControllerUri(), TEST_SCOPE, TEST_STREAM);
+
+        // won't be read because it's written after end poisitions are fetched
+        writer.writeEvent("onemore");
+
+        job = configureJob(conf, outputPath, "", endPos1);
+        status = job.waitForCompletion(true);
         Assert.assertTrue(job.isSuccessful());
 
-        File output = new File(outputPath.toUri() + "/");
-        Map<String, Integer> counts = getCounts(output);
+        output = new File(outputPath.toUri() + "/");
+        counts = getCounts(output);
 
-        Assert.assertEquals(new Integer(3), counts.get("pravega"));
+        Assert.assertEquals(null, counts.get("onemore"));
+        Assert.assertEquals(new Integer(1), counts.get("begin"));
+        Assert.assertEquals(new Integer(1), counts.get("pravega"));
         Assert.assertEquals(new Integer(1), counts.get("local"));
-        Assert.assertEquals(new Integer(2), counts.get("job"));
-        Assert.assertEquals(new Integer(3), counts.get("test"));
+        Assert.assertEquals(new Integer(1), counts.get("job"));
+        Assert.assertEquals(new Integer(1), counts.get("test"));
+        Assert.assertEquals(new Integer(1), counts.get("streamcut1"));
+        Assert.assertEquals(new Integer(1), counts.get("endonly"));
+
+        // TEST 2: with both start and end positions
+        writer.writeEvent("streamcut2 startandend");
+
+        fs = FileSystem.getLocal(conf);
+        if (fs.exists(outputPath)) {
+            fs.delete(outputPath, true);
+        }
+
+        String endPos2 = PravegaInputFormat.fetchLatestPositionsJson(
+                SETUP_UTILS.getControllerUri(), TEST_SCOPE, TEST_STREAM);
+
+        // won't be read because it's written after end poisitions are fetched
+        writer.writeEvent("twomore");
+
+        job = configureJob(conf, outputPath, endPos1, endPos2);
+        status = job.waitForCompletion(true);
+        Assert.assertTrue(job.isSuccessful());
+
+        output = new File(outputPath.toUri() + "/");
+        counts = getCounts(output);
+        Assert.assertEquals(3, counts.size());
+        Assert.assertEquals(new Integer(1), counts.get("streamcut2"));
+        Assert.assertEquals(new Integer(1), counts.get("startandend"));
+        Assert.assertEquals(new Integer(1), counts.get("onemore"));
+
+        // TEST 3: with start positions only
+        writer.writeEvent("streamcut3 startonly");
+
+        fs = FileSystem.getLocal(conf);
+        if (fs.exists(outputPath)) {
+            fs.delete(outputPath, true);
+        }
+
+        job = configureJob(conf, outputPath, endPos2, "");
```

Comment:
This usage pattern  (where the 4th Parameter `String endPos` is empty ) will not be used in a real world scenario, right?

(Reason: Even though `BatchClient#getSegments` returns the TAIL StreamCut there is is no way this can be returned from `PravegaInputFormat#getSplits()`)

Reply:
End user can set 4th parameter to empty if they want to process all events from start position to latest, and have no plan to run another job from current latest position. It's end user's choice, we just provide all options.